main.py

```diff
@@ -5,12 +5,15 @@
 import random
 import json
 import yaml
+import logging
 
 from pathlib import Path
 from typing import Dict, Any, Tuple, Callable, Sequence, Optional, List
 
 NUM_RETRIES = 3
 
+logging.basicConfig(level=logging.INFO)
```

Comment:
debug logs will be missed with this configuration of logging.
Consider changing to debug or adding a parameter to control level of logging

Reply:
let's have a debug logging by default for the beginning, after some time we can switch to info
and +1 for adding parameter to control the level